DOC: Add and fixup/move docs for descriptor changes (#25946)
numpy/numpy: doc/source/release/2.0.0-notes.rst

(`gh-25802 <https://github.com/numpy/numpy/pull/25802>`__)

Descriptor ``elsize`` and ``alignment`` access
----------------------------------------------
Unless compiling only with NumPy 2 support, the ``elsize`` and ``aligment``
fields must now be accessed via `PyDataType_ELSIZE`,

> `PyDataType_SET_ELSIZE`, and `PyDataType_ALIGNMENT`.

Reviewer: The links above are broken. Maybe you can fix the other warnings in the release notes, see the warnings starting [here](https://app.circleci.com/pipelines/github/numpy/numpy/25428/workflows/fbc25d8f-5616-44aa-9076-57dbecd94c1a/jobs/38702?invite=true#step-109-16587_106)

> In cases were the array is available, we advise using ``PyArray_ITEMSIZE``
> as it exists on all NumPy versions otherwise please see
> :ref:`_migration_c_descr`.

Reviewer: ```suggestion
In cases where the descriptor is attached to an array, we advise
using ``PyArray_ITEMSIZE`` as it exists on all NumPy versions.
Please see :ref:`_migration_c_descr` for more information.
```
